velox/functions/prestosql/DataSizeFunctions.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <map>
+#include <string>
+#include "velox/functions/Macros.h"
+#include "velox/type/DecimalUtil.h"
+
+namespace facebook::velox::functions {
+
+enum class Unit : int128_t {
+  BYTE = 1,
+  KILOBYTE = int128_t(1) << 10, // 2^10
+  MEGABYTE = int128_t(1) << 20, // 2^20
+  GIGABYTE = int128_t(1) << 30, // 2^30
+  TERABYTE = int128_t(1) << 40, // 2^40
+  PETABYTE = int128_t(1) << 50, // 2^50
+  EXABYTE = int128_t(1) << 60, // 2^60
+  ZETTABYTE = (int128_t(1) << 60) * (int128_t(1) << 10), // 2^70
+  YOTTABYTE = (int128_t(1) << 60) * (int128_t(1) << 20) // 2^80
+};
+
+inline Unit parseUnit(const std::string& unitString) {
+  static const std::map<std::string, Unit> unitMap = {
+      {"B", Unit::BYTE},
+      {"kB", Unit::KILOBYTE},
+      {"MB", Unit::MEGABYTE},
+      {"GB", Unit::GIGABYTE},
+      {"TB", Unit::TERABYTE},
+      {"PB", Unit::PETABYTE},
+      {"EB", Unit::EXABYTE},
+      {"ZB", Unit::ZETTABYTE},
+      {"YB", Unit::YOTTABYTE}};
+  auto it = unitMap.find(unitString);
+  VELOX_USER_CHECK(it != unitMap.end(), "Invalid data size: '{}'", unitString);
+  return it->second;
+}
+
+inline double parseValue(const std::string& value) {
+  try {
+    return std::stod(value);
+  } catch (const std::invalid_argument&) {
+    VELOX_USER_FAIL("Invalid data size: '{}'", value);
+  } catch (const std::out_of_range&) {
+    VELOX_USER_FAIL("Value out of range: '{}'", value);
```

Comment:
Can this one also be formatted like "Value out of range: '{}' ('{}B')"?

Reply:
Updated this